### PR TITLE
Add 'Year in Review' blog post

### PR DESCRIPTION
**Pull request by wints**
Similar to 2015's end-of-year writeup (http://flink.apache.org/news/2015/12/18/a-year-in-review.html), this post highlights some of the Flink community's accomplishments during the past year.

### CODE REVIEW

**Comment by wints** [comment]
@uce @rmetzger

File: _posts/2016-12-19-2016-year-in-review.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+### Github
+First, here's a summary of community statistics from <a href="https://github.com/apache/flink" target="_blank">GitHub</a>. At the time of writing:
+<ul>
+  <li><b>Contributors</b> have increased from 150 in December 2015 to 258 in December 2016 (up <b>72%</b>)</li>
+  <li><b>Stars</b> have increased from 813 in December 2015 to 1830 in December 2016 (up <b>125%</b>)</li>
+  <li><b>Forks</b> have increased from 544 in December 2015 to 1255 in December 2016 (up <b>130%</b>)</li>
+</ul>
+
+The community also welcomed <b>3 new committers in 2016</b>: Chengxiang Li, Greg Hogan, and Tzu-Li (Gordon) Lai.
```
Comment (rmetzger) [inline _posts/2016-12-19-2016-year-in-review.md]: Gordon's last name is Tai, not Lai: https://github.com/tzulitai

File: _posts/2016-12-19-2016-year-in-review.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+## Flink Forward 2016
+
+The <a href="http://2016.flink-forward.org/" target="_blank">second annual Flink Forward conference </a>took place in
+Berlin on September 12-14, and over 350 members of the Flink community came together for speaker sessions, training,
+and discussion about Flink. <a href="http://2016.flink-forward.org/program/sessions/" target="_blank">Slides and videos</a>
+ from speaker sessions are available online, and we encourage you to take a look if you’re interested in learning more
+ about how Flink is used in production in a wide range of organizations.
+
+Flink Forward will be expanding to <a href="http://sf.flink-forward.org/" target="_blank"> San Francisco in April 2017</a>
```
Comment (rmetzger) [inline _posts/2016-12-19-2016-year-in-review.md]: space in front of SF (effective double space)

**Comment by rmetzger** [comment]
Thank you for the great blog post!
+1 to merge once the typos are fixed.

**Comment by wints** [comment]
Thank you for the comments, @rmetzger! I've fixed typos and updated the release timeline image.

File: _posts/2016-12-19-2016-year-in-review.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+<ul>
+  <li><a href="https://aws.amazon.com/blogs/big-data/use-apache-flink-on-amazon-emr/" target="_blank">
+    Amazon EMR</a></li>
+  <li><a href="https://cloud.google.com/dataproc/docs/release-notes/service#november_29_2016" target= "_blank">
+    Google Cloud Dataproc</a></li>
+  <li><a href="https://www.lightbend.com/blog/introducing-lightbend-fast-data-platform" target="_blank">
+    Lightbend Fast Data Platform</a></li>
+</ul>
+
+In addition, the Flink community built an Apache Beam runner that, according to the Google team, is <a href="https://cloud.google.com/blog/big-data/2016/05/why-apache-beam-a-google-perspective"
```
Comment (fhueske) [inline _posts/2016-12-19-2016-year-in-review.md]: One could also argue that it was the Beam community who built the runner. Maybe rephrase to something like: "The Beam and the Flink communities teamed up to built a Flink runner for Apache Beam that, ..."?

File: _posts/2016-12-19-2016-year-in-review.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+to provide users with more control over how elements are evicted from a window.
+
+* **Fined-grained recovery from task failures**, as described in <a href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-1+%3A+Fine+Grained+Recovery+from+Task+Failures" target="_blank">FLIP-1</a>,
+to make it possible to restart only what needs to be restarted during recovery, building on cached intermediate results.
+
+* **Unified checkpoints and savepoints**, as described in <a href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-10%3A+Unify+Checkpoints+and+Savepoints" target="_blank">FLIP-10</a>, to
+allow savepoints to be triggered automatically--important for program updates for the sake of error handling because savepoints allow the user to modify both
+ the job and Flink version whereas checkpoints can only be recovered with the same job.
+
+* **Table API stream aggregations**, as described in <a href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-11%3A+Table+API+Stream+Aggregations" target="_blank">FLIP-11</a>, to
```
Comment (fhueske) [inline _posts/2016-12-19-2016-year-in-review.md]: I would rephrase to "Table API window aggregations, as described in FLIP-11 to support group-window and row-window aggregates on streaming and batch tables.

File: _posts/2016-12-19-2016-year-in-review.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+layout: post
+title:  "Apache Flink in 2016: Year in Review"
+date:   2016-12-19 09:00:00
+author: "Mike Winters"
+excerpt: "<p>As 2016 comes to a close, let's take a moment to look back on the Flink community's great work during the past year.</p>"
+categories: news
+---
+<p>2016 was an exciting year for the Apache Flink® community, and the
+  <a href="http://flink.apache.org/news/2016/03/08/release-1.0.0.html" target="_blank">release of Flink 1.0 in March</a>
+   marked the first time in Flink’s history that the community guaranteed backward compatibility for all
```
Comment (uce) [inline _posts/2016-12-19-2016-year-in-review.md]: Should we restrict this to _API_ backward compatability? Or are you explicitly referring to both API and savepoint compatability? As a side question: I've seen many non-native speakering committers use backward**s** compatability (with s). Are both OK or is the -s variant wrong? :smile:

File: _posts/2016-12-19-2016-year-in-review.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+<br><img src="{{ site.baseurl }}/img/blog/github-stats-2016.png" width="775" alt="Apache Flink GitHub Stats"/>
+<br>
+<br>
+
+Next, let's take a look at a few other project stats, starting with number of commits. If we run:
+
+```
+git log --pretty=oneline --after=12/31/2015 | wc -l
+```
+...inside the Flink repository, we'll see a total of **1839** commits so far in 2016.
```
Comment (uce) [inline _posts/2016-12-19-2016-year-in-review.md]: I think yesterday we passed the 10k commits mark, too. ;-)

File: _posts/2016-12-19-2016-year-in-review.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+ target="_blank">"sophisticated enough to be a compelling alternative to Cloud Dataflow when running on premise or on non-Google clouds"</a>.
+
+### Feature Timeline in 2016
+
+Here’s a selection of major features added to Flink over the course of 2016:
+
+<img src="{{ site.baseurl }}/img/blog/flink-releases-2016.png" width="775" alt="Flink Release Timeline 2016"/>
+
+If you spend time in the <a href="https://issues.apache.org/jira/browse/FLINK-4554?jql=project%20%3D%20FLINK%20AND%20issuetype%20%3D%20%22New%20Feature%22%20AND%20status%20%3D%20Resolved%20ORDER%20BY%20resolved%20DESC" target="_blank">Apache Flink JIRA project</a>, you’ll see that the Flink community has addressed every single one of the roadmap items identified
+in <a href="http://flink.apache.org/news/2015/12/18/a-year-in-review.html" target="_blank">2015’s year in review post</a>. Here's to making that an annual tradition :)
```
Comment (uce) [inline _posts/2016-12-19-2016-year-in-review.md]: Last sentence `.` missing?

**Comment by uce** [comment]
Looks great! +1 to merge after addressing the comments you find relevant.

File: _posts/2016-12-19-2016-year-in-review.md (hunk 5 above)
Comment (wints) [inline _posts/2016-12-19-2016-year-in-review.md]: I like the suggestion--I'll change the language to _API backward compatibility_. As for the side question: according to my sources, both are correct, but 'backward' is more common in US English and 'backwards' is more common in British English :)

File: _posts/2016-12-19-2016-year-in-review.md (hunk 4 above)
Comment (wints) [inline _posts/2016-12-19-2016-year-in-review.md]: Good suggestion, will update!

File: _posts/2016-12-19-2016-year-in-review.md (hunk 3 above)
Comment (wints) [inline _posts/2016-12-19-2016-year-in-review.md]: Good point--I've updated to use your suggested wording.

File: _posts/2016-12-19-2016-year-in-review.md (hunk 7 above)
Comment (wints) [inline _posts/2016-12-19-2016-year-in-review.md]: Updated!

File: _posts/2016-12-19-2016-year-in-review.md (hunk 6 above)
Comment (wints) [inline _posts/2016-12-19-2016-year-in-review.md]: Nice, I added that too.

**Comment by wints** [comment]
Thanks for the feedback, all! @fhueske @uce, I just pushed an updated version addressing your comments.

**Comment by fhueske** [comment]
thanks for the update. Merging this!
